File: Edit-Profile-Service/src/main/kotlin/com/kinok0/editprofileservice/repository/RefreshTokensRepository.java
package com.kinok0.editprofileservice.repository;

import com.kinok0.editprofileservice.entity.RefreshTokenEntity;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface RefreshTokensRepository extends CrudRepository<RefreshTokenEntity, UUID> {
    RefreshTokenEntity findRefreshTokensEntityByToken(String token);

    @Override
    List<RefreshTokenEntity> findAll();
}

File: Authentication-Service/src/main/kotlin/com/kinok0/authenticationservice/repository/RefreshTokensRepository.kt
package com.kinok0.authenticationservice.repository

import com.kinok0.authenticationservice.entity.RefreshTokenEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface RefreshTokensRepository : CrudRepository<RefreshTokenEntity, UUID> {
    fun findRefreshTokensEntityByToken(token: String): RefreshTokenEntity
    override fun findAll(): MutableList<RefreshTokenEntity>
}